app/src/main/java/com/tari/android/wallet/ui/fragment/onboarding/LocalAuthFragment.kt

```diff
@@ -0,0 +1,353 @@
+        animatorSet.addListener(object : AnimatorListenerAdapter() {
+            override fun onAnimationEnd(animation: Animator?) {
+                super.onAnimationEnd(animation)
+                doAuth()
+            }
+        })
+    }
+
+    private fun startAuthSuccessAnim() {
+        touchIdPromptContainer.visibility = View.VISIBLE
+        val touchIdBgFadeInAnim = ValueAnimator.ofFloat(0f, 1f)
+        touchIdBgFadeInAnim.addUpdateListener { valueAnimator: ValueAnimator ->
+            val alpha = valueAnimator.animatedValue as Float
+            touchIdPromptBg.alpha = alpha
+        }
+        touchIdBgFadeInAnim.duration = Auth.touchIdPromptDurationMs
+
+        val touchIdPromptFadeInAnim = ValueAnimator.ofFloat(0f, 1f)
+        touchIdPromptFadeInAnim.addUpdateListener { valueAnimator: ValueAnimator ->
+            val alpha = valueAnimator.animatedValue as Float
+            touchIdPrompt.alpha = alpha
+        }
+
+        touchIdPromptFadeInAnim.startDelay = Auth.touchIdPromptFadeInAnimDelayDuration
+        touchIdPromptFadeInAnim.duration = Auth.touchIdPromptDurationMs
+
+        val touchIdPromptFadeOutAnim = ValueAnimator.ofFloat(1f, 0f)
+        touchIdPromptFadeOutAnim.addUpdateListener { valueAnimator: ValueAnimator ->
+            val alpha = valueAnimator.animatedValue as Float
+            touchIdPrompt.alpha = alpha
+        }
+
+        touchIdPromptFadeOutAnim.startDelay = Auth.touchIdPromptFadeOutAnimDelayDuration
+        touchIdPromptFadeOutAnim.duration = Auth.touchIdPromptDurationMs
+
+        val touchIdBgFadeOutAnim = ValueAnimator.ofFloat(1f, 0f)
+        touchIdBgFadeOutAnim.addUpdateListener { valueAnimator: ValueAnimator ->
+            val alpha = valueAnimator.animatedValue as Float
+            touchIdPromptBg.alpha = alpha
+        }
+        touchIdBgFadeOutAnim.duration = Auth.touchIdPromptDurationMs
+
+        val fadeOutAnim = ValueAnimator.ofFloat(1f, 0f)
+        fadeOutAnim.addUpdateListener { valueAnimator: ValueAnimator ->
+            val alpha = valueAnimator.animatedValue as Float
+            authDescTextView.alpha = alpha
+            useAuthButton.alpha = alpha
+            authTypeImageView.alpha = alpha
+            smallGemImageView.alpha = alpha
+            authTitleTextContainer.alpha = alpha
+        }
+        fadeOutAnim.duration = Auth.removeFragmentDelayDuration
+
+        val animatorSet = AnimatorSet()
+        animatorSet.playSequentially(
+            touchIdBgFadeInAnim, touchIdPromptFadeInAnim,
+            touchIdPromptFadeOutAnim, touchIdBgFadeOutAnim, fadeOutAnim
+        )
+        animatorSet.addListener(object : AnimatorListenerAdapter() {
+            override fun onAnimationEnd(animation: Animator?) {
+                super.onAnimationEnd(animation)
+                startHomeActivity()
+            }
+        })
+        animatorSet.start()
+    }
+
+    private fun startHomeActivity() {
+        listener?.onAuthTypeSetSuccess()
```

Comment:
Not sure if the name reflects what's being done here. It's correct if we're setting the authentication type, but it's actually done at the OS settings level.